src/main/java/seedu/address/logic/commands/ChangeCommand.java

```diff
@@ -0,0 +1,208 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DURATION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TIME;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_APPOINTMENTS;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.appointment.Appointment;
+import seedu.address.model.appointment.Date;
+import seedu.address.model.appointment.Time;
+import seedu.address.model.patient.Patient;
+
+/**
+ * Edits the details of an existing appointment in the appointment book.
+ */
+public class ChangeCommand extends Command {
+
+    public static final String COMMAND_WORD = "change";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Reschedules an existing appointment for a patient "
+            + "by the index number used in the displayed appointment list. "
+            + "A new appointment will be created with the input values.\n"
+            + "The old appointment at the input index will be deleted from the appointment book.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_DATE + "DATE] "
+            + "[" + PREFIX_TIME + "TIME] "
+            + "[" + PREFIX_DURATION + "DURATION] \n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_DATE + "12-Dec-2021 "
+            + PREFIX_TIME + "4:00PM"
+            + PREFIX_DURATION + "30";
+
+    public static final String MESSAGE_EDIT_APPOINTMENT_SUCCESS = "Edited Appointment: %1$s";
+    public static final String MESSAGE_DUPLICATE_APPOINTMENT = "This appointment already exists in the appointment "
+            + "book.";
+
+    public static final String ASSIGNMENT_OVERLAP = "This time slot is occupied";
+    public static final String DATE_MISSING = "The date of appointment is missing";
+    public static final String TIME_MISSING = "The time of appointment is missing";
+    public static final String DURATION_MISSING = "The duration of appointment is missing";
+
+    private final Index index;
+    private final EditAppointmentDescriptor editAppointmentDescriptor;
+
+    /**
+     * @param index of the appointment in the filtered appointment list to edit
```

Comment:
Would a short description of the method be better?

Reply:
Added! Realize it was absent in EditCommand so I added it there as well.